app/models/manageiq/providers/redfish/physical_infra_manager/physical_server/provisioning.rb

```diff
@@ -0,0 +1,41 @@
+module ManageIQ::Providers::Redfish
+  module PhysicalInfraManager::PhysicalServer::Provisioning
+    def deploy_pxe_config(pxe_image, customization_template)
+      with_provider_object do |system|
+        if (macs = mac_addresses(system)).empty?
+          raise MiqException::MiqProvisionError, 'at least one MAC address is needed for provisioning'
+        end
+        macs.each { |mac| pxe_image.pxe_server.create_provisioning_files(pxe_image, mac, nil, customization_template) }
+      end
+    end
+
+    def reboot_using_pxe
+      with_provider_object do |system|
+        response = system.patch(
```

Comment:
@miha-plesko is this a call that can block for a period of time or does it return immediately?